### PR TITLE
core: Extract Pouch#byIdMaybe() from Merge

### PR DESCRIPTION
**Pull request by sebn**

### CODE REVIEW

File: core/merge.js [hunk 1]
```diff
@@ -503,16 +464,8 @@ class Merge {
...
-    try {
-      folder = await this.pouch.db.get(doc._id)
-    } catch (err) {
-      if (err.status === 404) {
-        return null
-      } else {
-        throw err
-      }
-    }
+    const folder /*: ?Metadata */ = await this.pouch.db.get(doc._id)
```
Comment (aenario) [inline core/merge.js]: Shouldn't this be `byIdMaybeAsync` ?

Comment (sebn) [reply core/merge.js]: - [ ] Definitely!

**Comment by sebn** [comment]
Fixed call & rebased on master. Waiting for CI.

@nono Can you think of some issue if we decide to throw on non-404 Pouch errors instead of logging a warning in Merge?

**Comment by nono** [comment]
Well, I don't see what errors can return PouchDB on a get (except 404). So, it's hard to tell.

**Comment by sebn** [comment]
I assume any other error would mean the system is currently broken in some way, so better throw instead of still trying to sync.

**Pull request merged**